File: GameLibrary/src/cc/lib/utils/NoDupsSet.java
package cc.lib.utils;

import java.util.HashSet;

/**
 * Created by Chris Caron on 4/26/22.
 */
public class NoDupsSet extends HashSet {

    @Override
    public boolean add(Object o) {
        if (contains(o))
            throw new IllegalArgumentException("Cannot add duplicate object:" + o);
        return super.add(o);
    }
}

File: GameLibrary2/src/cc/lib/utils/NoDupsSet.kt
package cc.lib.utils

/**
 * Created by Chris Caron on 4/26/22.
 */
class NoDupsSet : HashSet<Any?>() {
	override fun add(o: Any?): Boolean {
		require(!contains(o)) { "Cannot add duplicate object:$o" }
		return super.add(o)
	}
}
